src/websocket/client.ts

```diff
@@ -9,69 +9,75 @@ import { nanoid } from 'nanoid'
  * Uses Holochain's websocket WireMessage for communication.
  */
 export class WsClient {
-  socket: Websocket
-  pendingRequests: Record<string, { fulfill: Function }>
+  socket: Websocket;
+  pendingRequests: Record<string, { fulfill: Function }>;
 
   constructor(socket: any) {
-    this.socket = socket
-    this.pendingRequests = {}
+    this.socket = socket;
+    this.pendingRequests = {};
     // TODO: allow adding signal handlers later
   }
 
   emitSignal(data: any) {
     const encoded = msgpack.encode({
       type: 'Signal',
       data: msgpack.encode(data),
-    })
-    this.socket.send(encoded)
+    });
+    this.socket.send(encoded);
   }
 
   request<Req, Res>(data: Req): Promise<Res> {
-    const id = nanoid()
+    const id = nanoid();
     const encodedMsg = msgpack.encode({
       id,
       type: 'Request',
       data: msgpack.encode(data),
-    })
-    const promise = new Promise((fulfill) => {
-      this.pendingRequests[id] = { fulfill }
-    })
-    this.socket.send(encodedMsg)
-    return promise as Promise<Res>
+    });
+    const promise = new Promise(fulfill => {
+      this.pendingRequests[id] = { fulfill };
+    });
+    this.socket.send(encodedMsg);
+    return promise as Promise<Res>;
   }
 
   close(): Promise<void> {
     this.socket.close();
-    return this.awaitClose()
+    return this.awaitClose();
   }
 
   awaitClose(): Promise<void> {
-    return new Promise(resolve => this.socket.on('close', resolve))
+    return new Promise(resolve => this.socket.on('close', resolve));
   }
 
   static connect(url: string, signalCb?: Function): Promise<WsClient> {
     return new Promise((resolve, reject) => {
-      const socket = new Websocket(url)
+      const socket = new Websocket(url);
       socket.onopen = () => {
-        const hw = new WsClient(socket)
+        const hw = new WsClient(socket);
         socket.onmessage = (encodedMsg: any) => {
-          const msg = msgpack.decode(encodedMsg.data)
-          if (signalCb && msg.type === 'Signal') {
-            signalCb(msgpack.decode(msg.data))
-          } else if (msg.type === 'Response') {
-            const id = msg.id
-            if (hw.pendingRequests[id]) {
-              // resolve response
-              hw.pendingRequests[id].fulfill(msgpack.decode(msg.data))
+          encodedMsg.data.arrayBuffer().then((data: any) => {
```

Comment:
is there a type that we can refer to for `encodedMsg`, other than `any`?

was the change to using `encodedMsg.data.arrayBuffer()` necessary as part of the change to using `msgpack/msgpack` ? 

this would be the other core change I think?

Reply:
Yes exactly, @msgpack/msgpack does not accept a Blob, only arrayBuffers. Don't know about the any, I think ts cannot assume anything else.